### PR TITLE
Check for `authorization` field in mdns client

### PR DESCRIPTION
**Pull request by dannymeloy**
Uses the `api_auth` field in the mdns text records to determine which records to return

### CODE REVIEW

File: mdnsbridge/mdnsbridgeclient.py [hunk 1]
```diff
@@ -86,14 +82,16 @@ def getHrefWithException(self, srv_type, priority=None, api_ver=None, api_proto=
...
-    def _getValidServices(self, srv_type, priority, api_ver=None, api_proto=None):
+    def _getValidServices(self, srv_type, priority, api_ver=None, api_proto=None, api_auth=False):
         current_priority = 99
         valid_services = []
         for service in self.services[srv_type]:
             if api_ver is not None and api_ver not in service["versions"]:
                 continue
             if api_proto is not None and api_proto != service["protocol"]:
                 continue
+            if api_auth != service.get("authorization", False):
```
Comment (andrewbonney) [inline mdnsbridge/mdnsbridgeclient.py]: Looks like it'll work, but is there any reason for going against the existing pattern?

Comment (dannymeloy) [reply mdnsbridge/mdnsbridgeclient.py]: I think I was thinking backwards compatibility wise (on the off chance we were running an older version of nmoscommon)

Comment (andrewbonney) [reply mdnsbridge/mdnsbridgeclient.py]: Might be worth just checking we never call the method with some arguments missing. For api_ver/api_proto it looks like if the parameter is missing the method will return all services, whether they have the correct version/protocol or not. The auth addition doesn't seem to provide this option.

File: mdnsbridge/mdnsbridgeclient.py [hunk 2]
```diff
@@ -46,11 +46,11 @@ def getHref(self, srv_type, priority=None, api_ver=None, api_proto=None, api_aut
                 # Re-try after cache has been updated
                 return self.getHrefWithException(srv_type, priority, api_ver, api_proto, api_auth)
         except NoService:
-            self.logger.writeWarning("No DNS-SD service for for {}, priority={}, api_ver={}, api_proto={}".format(
+            self.logger.writeWarning("No DNS-SD service for {}, priority={}, api_ver={}, api_proto={}".format(
```
Comment (andrewbonney) [inline mdnsbridge/mdnsbridgeclient.py]: Sorry, only spotted this as a result of your typo fix. Could you add the value of api_auth to the logging here?

**Review by JamesGibo — APPROVED** [review]

**Review by andrewbonney — APPROVED** [review]

**Pull request merged**